pkg/status/status_server_test.go

```diff
@@ -74,6 +78,68 @@ func TestServer_getLoggerLevel(t *testing.T) {
 	}
 }
 
+func TestServer_setBpfLevel(t *testing.T) {
+	// Test in two modes
+	configs := []options.BpfConfig{{
+		Mode:        "ads",
+		BpfFsPath:   "/sys/fs/bpf",
+		Cgroup2Path: "/mnt/kmesh_cgroup2",
+	}, {
+		Mode:        "workload",
+		BpfFsPath:   "/sys/fs/bpf",
+		Cgroup2Path: "/mnt/kmesh_cgroup2",
+	}}
+
+	testLoggerLevelMap := map[string]int{
+		"error": constants.BPF_LOG_ERR,
+		"warn":  constants.BPF_LOG_WARN,
+		"info":  constants.BPF_LOG_INFO,
+		"debug": constants.BPF_LOG_DEBUG,
+	}
+	key := uint32(0)
+	actualLoggerLevel := uint32(0)
+	for _, config := range configs {
+		cleanup, bpfLoader := test.InitBpfMap(t, config)
+
+		server := &Server{
+			xdsClient: &controller.XdsClient{
+				WorkloadController: &workload.Controller{
+					Processor: nil,
+				},
+			},
+			bpfLogLevelMap: bpfLoader.GetBpfLogLevel(),
+		}
+
+		setLoggerUrl := patternLoggers
+		for logLevelStr, logLevelInt := range testLoggerLevelMap {
+			// We support both string and number
+			testLoggerLevels := []string{logLevelStr, strconv.FormatInt(int64(logLevelInt), 10)}
+			expectedLoggerLevel := uint32(logLevelInt)
+			for _, testLoggerLevel := range testLoggerLevels {
+				loggerInfo := LoggerInfo{
+					Name:  bpfLoggerName,
+					Level: testLoggerLevel,
+				}
+				reqBody, _ := json.Marshal(loggerInfo)
+				req := httptest.NewRequest(http.MethodPost, setLoggerUrl, bytes.NewReader(reqBody))
+				w := httptest.NewRecorder()
+				server.setLoggerLevel(w, req)
+
+				if w.Code != http.StatusOK {
+					t.Errorf("Expected status code %d, but got %d", http.StatusOK, w.Code)
+				}
+
+				server.bpfLogLevelMap.Lookup(&key, &actualLoggerLevel)
+
+				if actualLoggerLevel != expectedLoggerLevel {
+					t.Errorf("Wrong logger level, expected %d, but got %d", expectedLoggerLevel, actualLoggerLevel)
+				}
+			}
+		}
+		cleanup()
```

Comment:
So cleanup can not be called if error occurs above. maybe use t.Run(), we can use defer then